### ISSUE TITLE
DataReader - Incorretly processing ignored columns

### ISSUE THREAD

**Issue by VassilievVV** [jehugaleahsa/FlatFiles#70]
When schema is defined in code and contains IgnoredColumn - FlatFileDataReader includes IgnoredColumn in schema but does not include them in values.

```
        public static void TestDataReader()
        {
            const string data =
@"A,B,C
1,2,3
4,5,6";

            var schema = new SeparatedValueSchema();
            schema.AddColumn(new StringColumn("A"));
            schema.AddColumn(new IgnoredColumn());
            schema.AddColumn(new StringColumn("C"));

            var options = new SeparatedValueOptions()
            {
                IsFirstRecordSchema = true
            };

            using var textReader = new StringReader(data);
            var csvReader        = new SeparatedValueReader(textReader, schema, options);
            using var dataReader = new FlatFileDataReader(csvReader);

            var tblSchema = dataReader.GetSchemaTable();
            var strColumnNames = new StringBuilder();
            foreach (DataRow rowSchema in tblSchema.Rows)
            {
                if (strColumnNames.Length > 0)
                    strColumnNames.Append(", ");
                string strColumnName = Convert.ToString(rowSchema["ColumnName"]);
                if (string.IsNullOrWhiteSpace(strColumnName))
                    strColumnName = "\"\"";
                strColumnNames.Append(strColumnName);
            }

            Console.WriteLine($"Columns: {strColumnNames}.");

            while (dataReader.Read())
            {
                var values = dataReader.GetValues();

                var strValues = new StringBuilder();
                foreach (object value in values)
                {
                    if (strValues.Length > 0)
                        strValues.Append(", ");
                    string strValue = Convert.ToString(value);
                    if (string.IsNullOrWhiteSpace(strValue))
                        strValue = "\"\"";
                    strValues.Append(strValue);
                }

                Console.WriteLine($"Values:  {strValues}");
            }

            dataReader.Close();
        }
```

Sincerely yours,
Viatcheslav V. Vassiliev

**Comment by jehugaleahsa** [comment]
You're right and this caused some other bugs. I took your example here and converted it into two unit tests that helped me to resolve the issues. They should be fixed in the 4.12.0 release. Please give them a try at your first convenience and let me know. If everything looks good, please close this issue. Thanks!

**Comment by VassilievVV** [comment]
Thank you for quick fix and for good component.

Sincerely yours,
Viatcheslav V. Vassiliev
